hist/hist/inc/TH1.h

```diff
@@ -56,6 +60,80 @@ class TVirtualHistPainter;
 class TRandom;
 
 
+namespace{
+   std::vector<std::vector<int>> GetSliceIndices(const std::vector<std::pair<int, int>>& sliceEdges) {
+      const auto ndim = sliceEdges.size();
+      if (ndim == 0) return {};
+  
+      std::vector<std::vector<int>> slices(ndim);
+      for (size_t i = 0; i < ndim; ++i) {
+          for (int val = sliceEdges[i].first; val < sliceEdges[i].second; ++val) {
+              slices[i].push_back(val);
+          }
+      }
+  
+      size_t totalCombinations = 1;
+      for (const auto& slice : slices) {
+         totalCombinations *= slice.size();
+      }
+
+      std::vector<std::vector<int>> result(totalCombinations, std::vector<int>(3, 0));
+      for (size_t dim = 0; dim < slices.size(); ++dim) {
+         size_t repeat = 1;
+         for (size_t i = dim + 1; i < slices.size(); ++i) {
+               repeat *= slices[i].size();
+         }
+
+         size_t index = 0;
+         for (size_t i = 0; i < totalCombinations; ++i) {
+               result[i][dim] = slices[dim][(index / repeat) % slices[dim].size()];
+               ++index;
+         }
+      }
+
+      return result;
+  }
+}
+
+namespace ROOT::Internal {
+   /**
+    * @brief Creates a sliced copy of the given histogram.
+    * 
+    * @return A new histogram object that is a sliced version of the input histogram.
+    */
+   template <typename T>
+   T Slice(const T &histo, std::vector<Int_t>& args)
+   {  
+      static_assert(
+         std::is_pointer_v<decltype(histo.fArray)>,
+         "HistoType must expose fArray"
+      );
+
+      T slicedHisto = histo;
+      slicedHisto.SetDirectory(nullptr);
```

Comment:
Note the difference wrt this code
```cpp
TDirectory::TContext _{nullptr};
T slicedHisto = histo;
```
With the current code, there is still a registration first with `gDirectory`, then a deregistration afterwards. With this example instead, there is no registration at all since `gDirectory == nullptr` at the point of creation of the histogram.